src/provider/provider_os_memory.c

```diff
@@ -574,28 +574,32 @@ static umf_result_t os_alloc(void *provider, size_t size, size_t alignment,
                            : ALIGN_UP(os_provider->part_size, page_size);
     char *ptr_iter = addr;
 
-    do {
-        size_t s = bind_size < size ? bind_size : size;
-        ret = hwloc_set_area_membind(
-            os_provider->topo, ptr_iter, s, os_provider->nodeset[membind++],
-            os_provider->numa_policy, os_provider->numa_flags);
-
-        size -= s;
-        ptr_iter += s;
-        membind %= os_provider->nodeset_len;
-        if (ret) {
-            os_store_last_native_error(UMF_OS_RESULT_ERROR_BIND_FAILED, errno);
-            LOG_PERR("binding memory to NUMA node failed");
-            // TODO: (errno == 0) when hwloc_set_area_membind() fails on Windows,
-            // ignore this temporarily
-            if (errno != ENOSYS &&
-                errno != 0) { // ENOSYS - Function not implemented
-                // Do not error out if memory binding is not implemented at all
-                // (like in case of WSL on Windows).
-                goto err_unmap;
+    // Bind memory to NUMA nodes if numa_policy is other than DEFAULT
+    if (os_provider->numa_policy != HWLOC_MEMBIND_DEFAULT) {
```

Comment:
Do we need this if?
Calling once mbind it should not be an issue worth extra if. 
If we need it please move it few lines up - there is a code related to mbind

Reply:
Yes I want to avoid calling mbind at all and this can't be done without "if"

Reply:
done - moved